lib/mongo/auth/aws/request.rb

```diff
@@ -210,6 +213,38 @@ def authorization
           "AWS4-HMAC-SHA256 Credential=#{access_key_id}/#{scope}, SignedHeaders=#{signed_headers_string}, Signature=#{signature}"
         end
 
+        # Validates the credentials and the constructed request components
+        # by sending a real STS GetCallerIdentity request.
```

Comment:
This is interesting because this validation gives a much better error message to the user besides "authentication failed" but it is not part of the auth spec. I would feel more comfortable for someone on the security team to review this. If this is useful in Ruby perhaps we should open a drivers ticket to make all drivers perform this check.

Reply:
This validation is currently only used in the test suite. I intended to add a debugging option to the driver to make it perform this check prior to authenticating, but since this involves additional network roundtrips, and because it is possible for the application to not have network access to STS due to firewall rules (but the server would have STS access), I did not intend to have this check be performed by default. I can create a drivers ticket regarding this option if you like once aws auth implementation in Ruby is complete.

Reply:
Thanks for explaining. In that case, this LGTM. I'll leave the decision of whether to open a drivers ticket up to you.